log/format.go

```diff
@@ -98,212 +73,102 @@ type TerminalStringer interface {
-			case LvlCrit:
-				color = 35
-			case LvlError:
-				color = 31
-			case LvlWarn:
-				color = 33
-			case LvlInfo:
-				color = 32
-			case LvlDebug:
-				color = 36
-			case LvlTrace:
-				color = 34
-			}
+func TerminalFormat(r slog.Record, usecolor bool) []byte {
+	msg := escapeMessage(r.Message)
+	var color = 0
+	if usecolor {
+		switch r.Level {
+		case LevelCrit:
+			color = 35
+		case slog.LevelError:
+			color = 31
+		case slog.LevelWarn:
+			color = 33
+		case slog.LevelInfo:
+			color = 32
+		case slog.LevelDebug:
+			color = 36
+		case LevelTrace:
+			color = 34
 		}
+	}
 
-		b := &bytes.Buffer{}
-		lvl := r.Lvl.AlignedString()
-		if locationEnabled.Load() {
-			// Log origin printing was requested, format the location path and line number
-			location := fmt.Sprintf("%+v", r.Call)
-			for _, prefix := range locationTrims {
-				location = strings.TrimPrefix(location, prefix)
-			}
-			// Maintain the maximum location length for fancyer alignment
-			align := int(locationLength.Load())
-			if align < len(location) {
-				align = len(location)
-				locationLength.Store(uint32(align))
-			}
-			padding := strings.Repeat(" ", align-len(location))
-
-			// Assemble and print the log heading
-			if color > 0 {
-				fmt.Fprintf(b, "\x1b[%dm%s\x1b[0m[%s|%s]%s %s ", color, lvl, r.Time.Format(termTimeFormat), location, padding, msg)
-			} else {
-				fmt.Fprintf(b, "%s[%s|%s]%s %s ", lvl, r.Time.Format(termTimeFormat), location, padding, msg)
-			}
-		} else {
-			if color > 0 {
-				fmt.Fprintf(b, "\x1b[%dm%s\x1b[0m[%s] %s ", color, lvl, r.Time.Format(termTimeFormat), msg)
-			} else {
-				fmt.Fprintf(b, "%s[%s] %s ", lvl, r.Time.Format(termTimeFormat), msg)
-			}
-		}
-		// try to justify the log output for short messages
-		length := utf8.RuneCountInString(msg)
-		if len(r.Ctx) > 0 && length < termMsgJust {
-			b.Write(bytes.Repeat([]byte{' '}, termMsgJust-length))
-		}
-		// print the keys logfmt style
-		logfmt(b, r.Ctx, color, true)
-		return b.Bytes()
-	})
+	b := &bytes.Buffer{}
+	lvl := LevelAlignedString(r.Level)
+	if color > 0 {
+		fmt.Fprintf(b, "\x1b[%dm%s\x1b[0m[%s] %s ", color, lvl, r.Time.Format(termTimeFormat), msg)
+	} else {
+		fmt.Fprintf(b, "%s[%s] %s ", lvl, r.Time.Format(termTimeFormat), msg)
+	}
+	// try to justify the log output for short messages
+	length := utf8.RuneCountInString(msg)
+	if r.NumAttrs() > 0 && length < termMsgJust {
+		b.Write(bytes.Repeat([]byte{' '}, termMsgJust-length))
+	}
+	// print the keys logfmt style
+	logfmt(b, []slog.Attr{}, r, color, true)
+	return b.Bytes()
 }
 
 // LogfmtFormat prints records in logfmt format, an easy machine-parseable but human-readable
 // format for key/value pairs.
 //
 // For more details see: http://godoc.org/github.com/kr/logfmt
 func LogfmtFormat() Format {
-	return FormatFunc(func(r *Record) []byte {
-		common := []interface{}{r.KeyNames.Time, r.Time, r.KeyNames.Lvl, r.Lvl, r.KeyNames.Msg, r.Msg}
+	return FormatFunc(func(r slog.Record) []byte {
```

Comment:
So, it seems that the `func LogfmtFormat() Format` is not actually used, and can be just dropped, the _actual_ handler for `logfmt` is constructed thus: 
```golang
func LogfmtHandler(wr io.Writer) slog.Handler {
	return &builtinHandler{slog.NewTextHandler(wr, &slog.HandlerOptions{
		ReplaceAttr: builtinReplace,
	})}
}
```
Curious...

Reply:
IMO drop this method